packages/SwingSet/src/kernel/kernel.js

```diff
@@ -164,6 +163,18 @@ export default function buildKernel(kernelEndowments) {
     log: [],
   };
 
+  // This is a low-level output-only string logger used by old unit tests to
+  // see whether vats made progress or not. The array it appends to is
+  // available as c.dump().log . New unit tests should instead use the
+  // 'result' value returned by c.queueToExport()
```

Comment:
Do we have a picture of what unit tests will look like using that pattern?  I.e., are we going to want a helper function that looks like this anyway? I suspect a lot of tests are going to want the ability to accumulate messages in this manner, regardless of how the bundle of message strings gets passed back to the test driver.  I'm also mildly concerned that returning a value via `c.queueToExport()` will prove clunky.  It would probably be good to update a few tests to use the return paradigm so we can decide if we like it and start figuring out what kinds of test support scaffolding we should put in place to use it.

Reply:
Agreed. This style of test will look pretty different. OTOH it will make it a lot easier to exercise temporal checks, where you start an operation (and make sure only the first stage of consequences occur), let everything come to a halt, then submit a second operation (and look for the consequences), etc.

OT3H the `queueToExport` that you'd use to drive a test like that is pretty much limited to sending plain data (no references) to a named vat's default export, because it's unclean to try and predict any other object references that might be exported. OT4H, we've got plenty of tests that do just that (argv=mode -style) and they manage just fine.